pallets/pallet-nft/src/lib.rs

```diff
@@ -0,0 +1,244 @@
+use sp_runtime::DispatchResult;
+use sp_std::{prelude::*, vec::Vec};
+
+use primitives::{NftId, Role, FILENAME_MAXSIZE};
+use traits::ManageRoles;
+
+mod mock;
+mod tests;
+
+pub mod weights;
+pub use weights::*;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+/// Stores Nft data
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub struct NftData {
+	pub nft_id: NftId,
+	pub data: Vec<u8>,
+	pub file_name: Vec<u8>,
+}
+
+pub use module::*;
+
+#[frame_support::pallet]
+pub mod module {
+	use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		type WeightInfo: WeightInfo;
+
+		type RoleManager: ManageRoles<Self::AccountId>;
+
+		type EnsureGovernance: EnsureOrigin<<Self as frame_system::Config>::RuntimeOrigin>;
+
+		#[pallet::constant]
+		type MaxSize: Get<u32>;
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The Nft is not belong the user who want to deal with it.
+		Unauthorised,
+		/// The account role does not equal to the expected role.
+		IncorrectRole,
+		/// The Nft Id should exist.
+		InvalidNftId,
+		/// Data is too large.
+		DataTooLarge,
+		/// File name is too large.
+		FileNameTooLarge,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Created an Nft.
+		NftMinted { owner: T::AccountId, nft_id: NftId },
+		/// Burnt an Nft.
+		NftBurned { owner: T::AccountId, nft_id: NftId },
+		/// Transferred an Nft.
+		NftTransferred { from: T::AccountId, to: T::AccountId, nft_id: NftId },
+		/// An Nft created.
+		NFTPending { nft_id: NftId, file_name: Vec<u8> },
+		/// Auditor audited an nft pass or fail.
+		NftAudited { nft_id: NftId, approve: bool },
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn next_nft)]
+	pub type NextNftId<T: Config> = StorageValue<_, NftId, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn pending_nft)]
+	pub type PendingNft<T: Config> =
+		StorageMap<_, Blake2_128Concat, NftId, (NftData, T::AccountId)>;
+	#[pallet::storage]
+	#[pallet::getter(fn owners)]
+	pub type Owners<T: Config> = StorageMap<_, Blake2_128Concat, NftId, T::AccountId>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn nfts)]
+	pub type Nfts<T: Config> = StorageMap<_, Blake2_128Concat, NftId, NftData>;
+
+	#[pallet::pallet]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Create an Nft which pending for auditor checking.
+		#[pallet::call_index(0)]
+		#[pallet::weight(T::WeightInfo::request_mint())]
+		pub fn request_mint(
+			origin: OriginFor<T>,
+			file_name: Vec<u8>,
+			data: Vec<u8>,
+		) -> DispatchResult {
+			// Get the account id
+			let id = ensure_signed(origin)?;
+
+			// Ensure role is not auditor.
+			ensure!(T::RoleManager::role(&id) != Some(Role::Auditor), Error::<T>::IncorrectRole);
+
+			// Ensure nft data won't over flow.
```

Comment:
overflow is a specific term used when data > max size of a data type. 
Here is more like a max size limit that we artificially put in for our system.